src/FileReader.php

```diff
@@ -38,11 +33,13 @@ final class FileReader
      */
     public static function createSingleton()
     {
-        if (null === self::$instance) {
-            self::$instance = new self();
+        static $instance = null;
+
+        if (!$instance) {
```

Comment:
```suggestion
        if (null === $instance) {
```

Reply:
sorry @kubawerlos , i clicked "merge" and only after that I your comment popped up

Reply:
@keradus make an another PR then😉

Reply:
actually, there is no rule enforcing us to use any of the approaches. until we have the strong rule/check, enforcing it, we will always diverge. I see no value in raising PR proposing such change for a single place :shrug: